src/NuGet.Services.AzureSearch/NuGet.Services.AzureSearch.csproj

```diff
@@ -215,6 +215,9 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="NuGetGallery.Core">
+      <Version>4.4.5-master-2699519</Version>
```

Comment:
Explicitly adding a reference here to the latest version of `NuGetGallery.Core` to force the transitive dependency upgrade of `NuGet.Services.Validation.Common.Job`'s dependency on it.

I think we need to look into why `NuGet.Services.Validation.Common.Job` brings in the `NuGetGallery.Core` dependency. AFAICT, the only thing used from that dependency is some `IDiagnosticsService` related types and our cloud blob APIs and wrappers. Perhaps we could simplify things by extracting those into their own nupkg, e.g. `NuGet.Services.BlobStorage` or something.

There's really no need to drag along all of the other transitive dependencies that `NuGetGallery.Core` brings along, and we could untangle our nupkgs inter-dependencies by splitting up those APIs a bit more.

![image](https://user-images.githubusercontent.com/880728/58564154-b9452880-822c-11e9-90e8-31746f355994.png)

Reply:
This was intentional. There was a period when the churn in the validation jobs was extremely high (3 people working in parallel) and we used the NuGet.Services.Validation.Common.Job project to consolidate a lot of the duplicate code between job initialization (e.g. DI).

The config, storage, and entities context DI wire-up is shared in the project, part of this a dependency on NuGetGallery.Core. I think a better way of looking at this project name is "NuGet.Jobs.Common++". Yes it has bit of validation specific code but I think now that the high churn in validation is decreased, we should consider merging NuGet.Services.Validation.Common.Job down into NuGet.Jobs.Common. With this cross repo world we are in, the fewer projects we have the easier it is to maintain things.

I guess what I was getting at is that we've running into a lot of pain (some of which only found at runtime) when we mix package versions and have a lot of little libraries.

Anyhow, as long as runtime works for all of our entry points and there are no weird restore/build warnings, this is fine too.